pkg/pgmodel/querier/querier.go

```diff
@@ -274,64 +286,86 @@ func (q *pgxQuerier) queryMultipleMetrics(filter metricTimeRangeFilter, cases []
 
 // getMetricTableName gets the table name for a specific metric from internal
 // cache. If not found, fetches it from the database and updates the cache.
-func (q *pgxQuerier) getMetricTableName(metric string) (string, error) {
-	var err error
-	var tableName string
+func (q *pgxQuerier) getMetricTableName(schema, metric string) (string, string, string, error) {
+	var (
+		err             error
+		tableName       string
+		tableSchema     string
+		seriesTableName string
+	)
 
-	tableName, err = q.metricTableNames.Get(metric)
+	tableSchema, tableName, seriesTableName, err = q.metricTableNames.Get(schema, metric)
 
 	if err == nil {
-		return tableName, nil
+		return tableSchema, tableName, seriesTableName, nil
 	}
 
 	if err != errors.ErrEntryNotFound {
-		return "", err
+		return "", "", "", err
 	}
 
-	tableName, err = q.queryMetricTableName(metric)
+	tableSchema, tableName, seriesTableName, err = q.queryMetricTableName(schema, metric)
 
 	if err != nil {
-		return "", err
+		return "", "", "", err
+	}
+
+	// In case of unknown schema, we need to set it as a default one.
+	if schema == "" {
+		err = q.metricTableNames.SetDefaultSchema(tableSchema, metric, tableName, seriesTableName)
+
+		if err != nil {
+			return "", "", "", err
+		}
 	}
 
-	err = q.metricTableNames.Set(metric, tableName)
+	err = q.metricTableNames.Set(tableSchema, metric, tableName, seriesTableName)
```

Comment:
We set default schema for queries which don't specify the schema but we need this to store the cache with the actual schema so if somebody queries that specific schema, it will get a hit.

Reply:
If we want this logic to populate the cache twice, I think it should be inside the cache code.